src/full/Agda/Syntax/Parser/Lexer.x

```diff
@@ -54,13 +54,14 @@ $nonalpha    = $idchar # $alpha
 $white_notab = $white # \t
 $white_nonl  = $white_notab # \n
 
-@number       = $digit+ | "0x" $hexdigit+
 @prettynumber = $digit+ ([_] $digit+)*
               | "0x" $hexdigit+ ([_] $hexdigit+)*
               | "0b" $binarydigit+ ([_] $binarydigit+)*
 @integer      = [\-]? @prettynumber
-@exponent     = [eE] [\-\+]? @number
-@float        = @integer \. @number @exponent? | @number @exponent
+@decimal      = $digit+
+@exponent     = [eE] [\-\+]? @decimal
+@float        = [\-]? @decimal \. @decimal @exponent?
+              | [\-]? @decimal @exponent
```

Comment:
Your grammar seems to conform to Scala's (https://www.scala-lang.org/files/archive/spec/2.11/01-lexical-syntax.html#floating-point-literals) except that it does not allow something like `.01`.  Since the old grammar also did not allow it, I think we are good here.